### PR TITLE
added KFold splitting

### PR DESCRIPTION
**Pull request by lannguyen0910**
Hi @kaylode , i'm Lan Nguyen-Hoang. I saw your to-do-list earlier and i've worked on the "KFold splitting". Please review it and feel free to raise me an issue if anything was wrong.

### CODE REVIEW

File: utils/tools/split_kfold.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    X_train = df.iloc[:, :-1]
+    Y_train = df.iloc[:, -1]
+
+    cv = StratifiedKFold(n_splits=args.n_splits,
+                         random_state=args.seed, shuffle=args.shuffle)
+
+    for fold, (train_idx, val_idx) in enumerate(cv.split(X_train, Y_train)):
+
+        TRAIN = str(fold + 1) + '_train'
```
Comment (kaylode) [inline utils/tools/split_kfold.py:36]: Prefer fold's index starting with zero

File: utils/tools/split_kfold.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        VAL = str(fold + 1) + '_val'
+
+        train_df = df.iloc[train_idx].reset_index(drop=True)
+        val_df = df.iloc[val_idx].reset_index(drop=True)
+
+        os.makedirs(f'{args.out}/{fold + 1}')
+
+        train_csv_path = f'{args.out}/{fold + 1}/{TRAIN}.csv'
```
Comment (kaylode) [inline utils/tools/split_kfold.py:44]: Should use ```os.path.join``` or something like that to prevent bugs

File: utils/tools/split_kfold.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+from sklearn.model_selection import StratifiedKFold
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--csv', type=str, default='raw.csv',
+                    help='path to csv file')
+parser.add_argument('--seed', type=int, default=0,
+                    help='random seed (default: 0)')
+parser.add_argument('--out', type=str, default='.',
+                    help='directory to save the splits (default: .)')
+parser.add_argument('--n_splits', type=int, default=3,
```
Comment (kaylode) [inline utils/tools/split_kfold.py:14]: Prefer default num splits to be 5

**Review by kaylode — CHANGES_REQUESTED** [review]

File: utils/tools/split_kfold.py (hunk 3 above)
Comment (lannguyen0910) [reply utils/tools/split_kfold.py:14]: ...

File: utils/tools/split_kfold.py (hunk 2 above)
Comment (lannguyen0910) [reply utils/tools/split_kfold.py:44]: @kaylode yeah, at first i used os.path.join() in every f-string line, but it raises an error with double slashes

**Review by lannguyen0910 — COMMENTED** [review]
@kaylode i've change to match with your previous review. Please review it again if you have time.

**Review by kaylode — APPROVED** [review]

**Pull request merged**